upstairs/src/client.rs

```diff
@@ -1308,97 +1237,6 @@ impl DownstairsClient {
         up_state: &UpstairsState,
         ddef: &mut RegionDefinitionStatus,
     ) -> Result<NegotiationResult, NegotiationError> {
-        /*
```

Comment:
It's a little surprising to see this entire block comment get deleted, at least it could be broken up and the correct bits moved somewhere appropriate?

Reply:
Yeah, it's tangential to the main body of the PR, but the comment was significantly out of date / incorrect.  We've basically moved all of the relevant info into the [`NegotiationState` docstring](https://github.com/oxidecomputer/crucible/blob/34a5d8722fdaaef364aed07b58f0c9dc712fe06f/upstairs/src/client.rs#L1901-L1942) instead.